### PR TITLE
(Tentative) Modernize extension

### PR DESCRIPTION
**Pull request by jonesbusy**
This doesn't work. But I don't have more time to invest on it for the moment.

FYI @sridamul this was just my tentative some weeks ago. I think we are not far, but I'm probably missing some things.

The circular disapear but i think I have some extension excluded by mistake that prevent Jenkins to start

### Testing done

```[tasklist]
### Submitter checklist
- [ ] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [ ] Ensure that the pull request title represents the desired changelog entry
- [ ] Please describe what you did
- [ ] Link to relevant issues in GitHub or Jira
- [ ] Link to relevant pull requests, esp. upstream and downstream changes
- [ ] Ensure you have provided tests - that demonstrates feature works or fixes the issue
```

### CODE REVIEW

**Comment by sridamul** [comment]
I will try to figure it out in the next couple of hours. Thanks for your work, @jonesbusy.

File: src/main/java/org/jenkinsci/plugins/ConfigurableExtensionFilter.java [hunk 1]
```diff
@@ -14,37 +16,17 @@
...
-        @Override
-        public boolean filter(@CheckForNull Object context, @Nonnull Descriptor descriptor) {
-            Class contextClass = context == null ? null : context.getClass();
-            return DESCRIPTOR.allows(contextClass, descriptor.getClass().getName());
-        }
-    };
-
-    @Extension
-    public static final DescriptorImpl DESCRIPTOR = new DescriptorImpl();
+    public static /*almost final*/ DescriptorImpl DESCRIPTOR;
```
Comment (sridamul) [inline src/main/java/org/jenkinsci/plugins/ConfigurableExtensionFilter.java:19]: `DESCRIPTOR should be package protected`. So, I would recommend to remove public

File: src/main/java/org/jenkinsci/plugins/ConfigurableExtensionFilter.java [hunk 2]
```diff
@@ -14,37 +16,17 @@
...
 
     @Symbol("configurableExtensionFilter")
+    @Extension
     public static final class DescriptorImpl extends Descriptor<ConfigurableExtensionFilter> {
```
Comment (sridamul) [inline src/main/java/org/jenkinsci/plugins/ConfigurableExtensionFilter.java:24]:
```
static {
            DESCRIPTOR = new DescriptorImpl();
        }
```
initialize DESCRIPTOR field before it is accessed

**Comment by sridamul** [comment]
Hey @jonesbusy, I have made a few changes that seem to work perfectly in the local environment. Also please test in your local environment if possible

**Comment by sridamul** [comment]
https://github.com/jenkinsci/extension-filter-plugin/assets/107404972/7bb8b9ee-b93e-448f-b22e-dd6529b09380

also the circular disappeared as mentioned by you

File: src/main/java/org/jenkinsci/plugins/ConfigurableExtensionFilter.java (hunk 2 above)
Comment (jonesbusy) [reply src/main/java/org/jenkinsci/plugins/ConfigurableExtensionFilter.java:24]: I think this need some test. My first feeling is that Jenkins will load all extension even if they are ignored.

We need to test disabling an extension (for example freestyle job) and see if they persist after a reload and reboot.

Perhaps also a test with RestartableJenkinsRule can validate this.

If you are motivated you can take the code from this PR and update the code based and your comment and provide test if possible

**Pull request closed without merging**